### ISSUE TITLE
WLoC, World List of Cycads (id 1163): test report

### ISSUE THREAD

**Issue by yroskov** [CatalogueOfLife/testing#267]
The export received from Michael Calonje on 2024-07-11 as _World_List_of_Cycads_Export20240711_190713.xlsx_ export file from BRAHMS; plus _Genera_Export20240712_110746.xlsx_ and _Families_Export20240712_110714.xlsx_ with authors for genera and families.

Export files converted by @mdoering. Imported on DEV https://www.dev.checklistbank.org/dataset/1163/imports

**Comment by yroskov** [comment]
**Imported on DEV 2024-07-24, iteration 1** https://www.dev.checklistbank.org/dataset/1163/imports 

Metrics:

![image](https://github.com/user-attachments/assets/ac616aed-f998-4c5c-b5b2-13fdaf97e6a6)

- [ ] Imported: 394 spp (vs 355 spp of Feb 2019) 

- [x] Metadata:  alias applied as **WLoC** (before cycads list had no short name in CoL; version corrected as Jul 2024 / 2024-07-11. Other metadata were copied from previous version.

- [x] Classification: ranks order (1) - family (2) - genus (10) - species (394) - **infraspecific names** (35)
- [x] **Infraspecific names need to be split into ranks subspecies, variety & forma.** Fields in the source file: Subspecies |SubspeciesAuthorName	| Variety | VarietyAuthorName | Forma | FormaAuthorName = FIXED in Iteration 3 

- [x] Families and genera have authorstrings

- [x] Shall we include in the CoL names with _TaxStatus_  **Invalid** & **NomenDubium** (other values Accepted & Synonym): yes, if they have parent accepted name; no (bare name), if they have no parent accepted name = _need to be confirmed with Michael_. In the iteration 2024-07-24, NomenDubium names got a status "prov acc".
See also field _Invalid_ with values **inv** & blank. Sometimes **inv** in _Invalid_ does not equal to **invalid** in _TaxStatus_
= FIXED IN ITERATION 3

_Note:_ number in square brackets in _CalcFullName_ in the source file indicates homonyms (e.g. Ceratozamia fuscoviridis [1] & Ceratozamia fuscoviridis [2])

- [x] **Binomials are incorrectly placed as synonyms of the genus** (e.g. Eriozamia hort ex J.Schust. (nomen invalidum), Eriozamia latifolia Hort. ex J.Schust. (nomen invalidum), Palmifolium Kuntze (nomen legitimum), Zamia amplifrons Hort. ex J.Schust. (nomen invalidum), Zamia brachyphylla Hort. ex J.Schust. (nomen invalidum) etc. **are synonyms of Zamia L.**)
These are _mainly_ names with status _nomen invalidum_ (field _TaxStatus_ with value _Invalid_)
= FIXED in Iterations 2-3
https://www.dev.checklistbank.org/dataset/1163/taxon/41b2a097-b108-4852-9ce1-c39ddfb0b25f
![image](https://github.com/user-attachments/assets/f5a2d41f-0496-46e2-8571-e3790705d588)
Almost all genera have binomials in synonymy.

- [ ] Sector: 
- [ ] Extinct flag: n/a

**Comment by mdoering** [comment]
I have just uploaded a new version which improves handling of nomen dubium, invalid & illegitimate names.
I will stop here and leave it as it is during my absence. So you can review 

Now all accepted names are placed in the genus, incl infraspecific names.
Doubtful and invalid names are treated as bare names, unless they have an explicit synof relationship (which is rare)

**Comment by yroskov** [comment]
Thanks! I'll continue with a new iteration.

Seems, proper tests and consultations with WLoC will take a time. Have a nice vacations!

**Comment by mdoering** [comment]
I see infraspecifc names lack a proper rank. I will fix that quickly and upload. 5 minutes...

**Comment by mdoering** [comment]
searching for [tridentata](https://www.dev.checklistbank.org/dataset/1163/names?facet=rank&facet=issue&facet=status&facet=nomStatus&facet=nomCode&facet=nameType&facet=field&facet=authorship&facet=authorshipYear&facet=extinct&facet=environment&facet=origin&facet=sectorMode&facet=secondarySourceGroup&facet=sectorDatasetKey&facet=group&limit=50&offset=0&q=tridentata&sortBy=taxonomic) gives you a nice idea of bare names vs synonyms

**Comment by yroskov** [comment]
**Imported on DEV 2024-07-24, iteration 3**

Metrics:

![image](https://github.com/user-attachments/assets/565ef66b-f1ac-4609-bbd3-ff63390ddb45)


- [x] Imported: 376 spp (vs 394 spp of 2024-07-24 iteration 1) 

- [x] Classification: ranks order (1) - family (2) - genus (10) - species (376) - subspecies (12) - variety (2) - (no _accepted_ forms in the source file)

- [x] Extinct flag: n/a
- [x] Distribution: OK
- [x] Link to the species page in the original website

**Comment by yroskov** [comment]
**Iteration 3** (imported on DEV 2024-07-24, 4:14 PM ) summary of remaining issues

- [x] ~~All **Invalid** & **NomenDubium** names in _TaxStatus_ should have "bare name" status in CoL~~.
I have checked 10 remaining **Invalid** names (9 spp & 1 var) which have status "synonym" in the Iteration 3 against WLoC website: these names are not included in synonymy; they listed separately as _Invalid Name Publication_ = IT'S WEBSITE INTERPRETATION ONLY

= After discussion with Michael: **KEEP THESE 10 in COL AS THEY ARE NOW IN ITERATION 3** (i.e. as synonyms)

_Example:_ 
Encephalartos caffer (Thunb.) Lehm. in DEV https://www.dev.checklistbank.org/dataset/1163/taxon/8784b87c-e197-4aea-bb62-8b55abae668d 

Encephalartos caffer (Thunb.) Lehm. in WLoC http://cycadlist.org/taxon.php?Taxon_ID=280
Encephalartos cycadis Sweet INVALID in WLoC http://cycadlist.org/taxon.php?Taxon_ID=624

Here is a list of 10 names:
![image](https://github.com/user-attachments/assets/45ed647b-7bd8-409a-8962-453ab6244245)

Zamia nana [2]
Macrozamia pectinata
Macrozamia littoralis
Encephalartos villosus var. nobilis
Encephalartos douglasii
Macrozamia hopei [1]
Encephalartos pumilus
Zamia parasitica
Encephalartos latifolius
Encephalartos cycadis

**Comment by yroskov** [comment]
**Iteration 3** (imported on DEV 2024-07-24, 4:14 PM ) summary of remaining issues

- [x] References: **missing Author(s) and Year**. For plants, we usually create CoL bibliographic reference from the "nomenclatural citation" (fields _SpeciesAuthorName_ (in the case of combination, we take only author(s) after right bracket only (i.e. (Salisb.) **Miq.**) ) + _PublishYear_ + _ProtologueCitation_). Usually, I also ask to add preposition with colon **In:** in the Source/Title. For example, Bowenia serrulata (W.Bull) Chamb. - Ref **Chamb. In: Bot. Gaz. 54: 419. (1912).** https://www.catalogueoflife.org/data/taxon/5WQC2 = FIXED in prod version of 2024-08-14
![image](https://github.com/user-attachments/assets/18524135-180b-4690-95e5-74fb47a1096e)
![image](https://github.com/user-attachments/assets/5f979b73-bdd1-4fff-8d5c-87339fc0d3a2)

![image](https://github.com/user-attachments/assets/6b4c46ad-39fc-4330-bb89-62511808cae7)

**Comment by yroskov** [comment]
**Iteration 3** (imported on DEV 2024-07-24, 4:14 PM ) summary of remaining issues

ISSUES

![image](https://github.com/user-attachments/assets/5d59fb60-b862-4ae4-a4f5-7c575e024bd5)

**Comment by yroskov** [comment]
**Iteration 3** (imported on DEV 2024-07-24, 4:14 PM ) summary of remaining issues

ISSUES

- [x] _Parent Species Missing_ & _Accepted Name Missing_, 14: subspecies which are not assigned to the parent species. _Links between subspecies and parent species are missing_  = FIXED in prod version of 2024-08-14
https://www.dev.checklistbank.org/dataset/1163/names?facet=rank&facet=issue&facet=status&facet=nomStatus&facet=nomCode&facet=nameType&facet=field&facet=authorship&facet=authorshipYear&facet=extinct&facet=environment&facet=origin&facet=sectorMode&facet=secondarySourceGroup&facet=sectorDatasetKey&facet=group&issue=parent%20species%20missing&limit=50&offset=0

In the Tree, they appear in the parent genus, i.e. outside existing species. For example:
https://www.dev.checklistbank.org/dataset/1163/classification?taxonKey=210f828b-069f-456b-9afc-d82421a62ae4
![image](https://github.com/user-attachments/assets/379c3462-0989-4911-847e-e4e3ae3b82a5)

**Comment by mdoering** [comment]
I have fixed the missing species issue for infraspecies, but there are still synonyms missing an accepted name.
The original data is lacking some information, e.g. [Cycas longipetiolula](https://www.dev.checklistbank.org/dataset/1163/name/87be5c92-fcaa-4a06-b6e7-ef37c1e6e2b9) claims to have the accepted name `Cycas bifida x c. multipinnata` with id e781d518-6d82-4dfb-be67-893e1d026cff, [both which do not exist](https://www.dev.checklistbank.org/dataset/1163/verbatim?q=e781d518-6d82-4dfb-be67-893e1d026cff) in the spreadsheet. Not much I can do here.

The original website does have that info: http://cycadlist.org/taxon.php?Taxon_ID=151

Another example of a missing species referenced from synonyms is 
> 5eafe3fa-bee5-4fa3-be3f-480cb7c08ff1	Phoenix dactylifera

**Comment by mdoering** [comment]
For the references I am uncertain if adding the nomenclatural author is improving the data. It is not a botanical practice you find on the [source site](http://cycadlist.org/taxon.php?Taxon_ID=9), [IPNI](https://www.ipni.org/n/296944-1), [WFO](https://wfoplantlist.org/taxon/wfo-0000570066-2024-06?page=1) or others.

Does it hurt to leave it as it is?

**Comment by mdoering** [comment]
I have uploaded the latest version with no reference change to prod - where I only updated the version/issued date metadata

**Comment by yroskov** [comment]
It is CoL standard for building bibliographic reference from nomenclatural citation in the botanical scientific name. If you need help on how to do it, please talk to @gdower. He completed this for many botanical GSDs.

**Comment by yroskov** [comment]
>I have uploaded the latest version

Thank you, Markus!

**Comment by mdoering** [comment]
ok, I have tried my best to also change the references. It's on prod already.

**Comment by mdoering** [comment]
Damn, there are some problems with the references. Going to fix that ...
https://www.checklistbank.org/dataset/1163/references?limit=50&offset=0

**Comment by mdoering** [comment]
Now they look good to me: https://www.checklistbank.org/dataset/1163/references?limit=50&offset=0
I stop here

**Comment by yroskov** [comment]
Wow! Perfect. Many thanks!

**Comment by yroskov** [comment]
>IPNI, WFO or others

Yeah, it would be nice if all these resources send us properly prepared bibliographic references (and with page/publicaion URLs). Unfortunately, most of them are far behind their users expectations. That's why we at CoL do our best to improve their data slightly and temporarily.

**Comment by yroskov** [comment]
**WLoC: World List of Cycads of 2024-07-11**; imported 2024-08-14 (converted by Markus)

- [x] Imported: 376 spp (vs 355 spp of Feb 2019)
- [x] Metadata:  Alias added as **WLoC** (see email exchange with Michael Calonje); version applied as **Jul 2024 / 2024-07-11**. Other metadata were saved from previous version.
- [x] Classification: OK. 
- [x] Sector: broken (previous sector was Class Cycadopsida). **Order Cycadales** re-established as a sector 2024-08-14 via editing Target & Subject in _Sector_ https://www.checklistbank.org/catalogue/3/sector?limit=100&offset=0&subjectDatasetKey=1163
![image](https://github.com/user-attachments/assets/399e6c8b-2c5b-44b9-8bc9-9f8b1fd44416)
- [x] flag "extinct": n/a
- [ ] DOI: 

Metrics

![image](https://github.com/user-attachments/assets/b9c01c6e-a0c4-4926-a0ee-18b770fadc37)

ISSUES assessed 2024-08-15

![image](https://github.com/user-attachments/assets/fe04d695-e394-46c7-ac6f-ae8c46b79232)

TASKS

![image](https://github.com/user-attachments/assets/7042d8d3-9997-4d74-a970-ab766cab0b1a)

Resolved 2024-08-15:

![image](https://github.com/user-attachments/assets/850c1700-eb0e-4a3e-8b6d-6deab87f831a)

**Synced 2024-08-15**

See colon in Target - will this affect results of sync?

![image](https://github.com/user-attachments/assets/a4fd674e-9e93-4676-8647-2eab79c3420b)

**Comment by yroskov** [comment]
So far, deletion of the sector failed: "with status code 503. The SyncManager is currently not available"

![image](https://github.com/user-attachments/assets/5c45cdf0-67f6-4edc-99fe-4ce778754870)

**Comment by mdoering** [comment]
yes, see slack. I had to do some urgent changes in the db and stop all write operations for the next 1-2 hours probably.
Wasn't expected, sorry

**Comment by yroskov** [comment]
- [x] Sector: See colon in Target - will this affect results of sync? = Yes, order Cycadales is placed in the top of the tree:

![image](https://github.com/user-attachments/assets/4f055a7a-e093-48da-81bd-8ee1df7df6bb)

Resolved:

Step 1. Delete sector in full. Node disappeared in the Tree, but software says _Request failed with status code 404. Sector 3:98 does not exist. HTTP method: GET. https://api.checklistbank.org/dataset/3/sector/98_
Step 2. Establish class Cycadopsida Brongn. (it was vanished)
Step 3. Establish order Cycadales inside Cycadopsida as a sector
 
**Re-synced 2024-08-15**
